### DISCUSSION TITLE
Foreground Jobs on iOS Device: Can't get it to trigger

### DISCUSSION THREAD

**Discussion by XMLDiva** [shinyorg/shiny#1034] [Q&A] [answered]
I have code in the startup to use jobs and trigger a foreground job. My understanding of foreground jobs is that it creates a timer to hit the job while the app is open. I've debug deployed to a physical device, but it doesn't trigger the job on app load. It looks like it'll trigger sporadically in the background on iOS schedule rather than a foreground process.

Do foreground jobs run with a timer for iOS, or do they hook into the iOS's background job scheduling?

Currently using Shiny / Shiny.Core / Shiny.Jobs 2.5.4

`services.UseJobs();

services.RegisterJob(new JobInfo(typeof(SyncJob), "Job", true)
	{
		RunOnForeground = true, 
		Repeat = true,
		PeriodicTime = new TimeSpan(0, 3, 0)
	});
`

**Comment by aritchie** [comment, marked as answer]
It won't trigger the job on app start.  It triggers it according to a timer set.  The application has to remain in the foreground for X seconds (default 30 seconds) before a foreground job will be triggered.  You can find the logic here:

https://github.com/shinyorg/shiny/blob/v2.5.0/src/Shiny.Jobs/Infrastructure/JobLifecycleTask.cs

**Reply by XMLDiva** [reply]
Thanks! For whatever reason, I always forget to switch to 2.5 branch so I appreciate the link.

I was playing around with it and I got it to trigger every 30 seconds once but then couldn't get it to do again with more attempts. I'll have to dig in it further (this is lower priority right now but will be pushed up eventually in what I'm working on). If it's something with Shiny, I'll create a bug for it. But the more likely case, if it's something in my code, I'll just learn something new.
